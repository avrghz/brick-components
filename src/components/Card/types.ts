export const SHADOW = ['never', 'always', 'hover'] as const

export type Shadow = typeof SHADOW[number]
