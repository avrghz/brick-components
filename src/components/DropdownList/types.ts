export interface Option {
    label: string
    value: string
    disabled?: boolean
}

export type Callback = (option: Option) => void
