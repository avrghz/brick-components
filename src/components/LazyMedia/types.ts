export type MediaType = 'img' | 'picture'
