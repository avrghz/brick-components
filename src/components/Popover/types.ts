export const TRIGGER_ON = ['click', 'focus', 'hover', 'manual'] as const

export type TriggerOn = typeof TRIGGER_ON[number]
