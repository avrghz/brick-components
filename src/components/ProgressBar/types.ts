export const BAR_TYPE = ['line', 'circle', 'dashboard'] as const

export type BarType = typeof BAR_TYPE[number]

export interface ProgressProps {
    progress: number
    showPercentageText?: boolean
    textInside?: boolean
}

export interface ProgressTextProps extends Pick<ProgressProps, 'progress'> {
    show?: boolean
    isInnerText?: boolean
}

export interface ProgressCircleProps extends ProgressProps {
    strokeDash: [number, number]
    strokeDashOffset: number
    flip?: boolean
}
