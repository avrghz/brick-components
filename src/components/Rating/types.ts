export interface StarProps {
    hover?: boolean
    activeColor?: string
}

export interface Colors {
    [key: number]: string
}
