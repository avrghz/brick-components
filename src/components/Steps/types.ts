import { JSXBase } from '@stencil/core/internal'

export const STEP_STATE = ['process', 'wait', 'success', 'error', 'finish'] as const

export interface StepComponent {
    step: number
    title: string
    description?: string
    state?: typeof STEP_STATE[number]
}

export interface StepProps extends StepComponent {
    style?: JSXBase.HTMLAttributes<HTMLDivElement>['style']
    isLast?: boolean
    isCentered?: boolean
}
