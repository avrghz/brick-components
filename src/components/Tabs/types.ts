export const POSITION = ['top', 'bottom', 'left', 'right'] as const

export const VARIANT = ['simple', 'card', 'border-card'] as const

export type Position = typeof POSITION[number]

export type Variant = typeof VARIANT[number]
