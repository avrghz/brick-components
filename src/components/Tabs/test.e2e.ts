import { newE2EPage, E2EPage, E2EElement } from '@stencil/core/testing'
import { VARIANT, POSITION } from './types'

describe('<bk-tabs/>', () => {
    let page: E2EPage
    let component: E2EElement

    const getTab = async (id: string) => await component.find(`bk-tab-header[tab="${id}"]`)

    beforeEach(async () => {
        page = await newE2EPage()
        await page.setContent(`
        <bk-tabs>
            <div slot="header">
                <bk-tab-header tab="tab_1" active="true">Tab 1</bk-tab-header>
                <bk-tab-header tab="tab_2">Tab 2</bk-tab-header>
            </div>
            <bk-tab-panel tab="tab_1">Content 1</bk-tab-panel>
            <bk-tab-panel tab="tab_2">Content 2</bk-tab-panel>
        </bk-tabs>`)
        component = await page.find('bk-tabs')
    })

    it('should render', async () => {
        expect(component).toMatchSnapshot(`snapshot_overall`)
    })

    it('should change tab on click', async () => {
        const tabs = ['tab_2', 'tab_1']

        await (await getTab(tabs[0])).click()
        expect(component).toMatchSnapshot(`snapshot_${tabs[0]}_click`)

        await (await getTab(tabs[1])).click()
        expect(component).toMatchSnapshot(`snapshot_${tabs[1]}_click`)
    })

    describe('@Prop variant', () => {
        it(`if not provided, should render as simple variant`, async () => {
            expect(component).toHaveClass(`bk-tabs--simple`)
        })

        VARIANT.forEach((v) => {
            it(`should render as ${v} variant`, async () => {
                component.setProperty('variant', v)
                await page.waitForChanges()
                expect(component).toHaveClass(`bk-tabs--${v}`)
            })
        })
    })

    describe('@Prop position', () => {
        it(`if not provided, should render as top positioned`, async () => {
            expect(component).toHaveClass(`bk-tabs--top`)
        })

        POSITION.forEach((p) => {
            it(`should render as ${p} positioned`, async () => {
                component.setProperty('position', p)
                await page.waitForChanges()
                expect(component).toHaveClass(`bk-tabs--${p}`)
            })
        })
    })

    describe('@Event bkSelect', () => {
        it('should fire bkSelect event, on tab selection', async () => {
            const id = 'tab_2'
            const bkSelect = await component.spyOnEvent('bkSelect')
            await (await getTab(id)).click()
            expect(bkSelect).toHaveReceivedEventDetail(id)
        })
    })
})
