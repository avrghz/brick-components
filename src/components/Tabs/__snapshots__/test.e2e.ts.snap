// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bk-tabs/> should change tab on click: snapshot_tab_1_click 1`] = `
<bk-tabs aria-orientation="horizontal" class="bk-tabs bk-tabs--simple bk-tabs--top hydrated">
  <!---->
  <div class="bk-tabs__header">
    <div class="bk-tabs__scroll">
      <div class="bk-tabs__nav" role="tablist">
        <div class="bk-tabs__active-bar" style="transform: translateX(0px); width: 34.25px;"></div>
        <div slot="header">
          <bk-tab-header active="" aria-controls="pane_tab_1" aria-selected="true" class="bk-tab-header hydrated is-active" id="tab_1" role="tab" tab="tab_1" tabindex="0">
            <!---->
            <div data-tab-header>
              Tab 1
            </div>
          </bk-tab-header>
          <bk-tab-header aria-controls="pane_tab_2" aria-selected="false" class="bk-tab-header hydrated" id="tab_2" role="tab" tab="tab_2" tabindex="-1">
            <!---->
            <div data-tab-header>
              Tab 2
            </div>
          </bk-tab-header>
        </div>
      </div>
    </div>
  </div>
  <bk-tab-panel aria-labelledby="tab_1" class="bk-tab-panel hydrated" id="pane_tab_1" role="tabpanel" tab="tab_1">
    <!---->
    Content 1
  </bk-tab-panel>
  <bk-tab-panel aria-labelledby="tab_2" class="bk-tab-panel hydrated" id="pane_tab_2" role="tabpanel" tab="tab_2" style="display: none;">
    <!---->
    Content 2
  </bk-tab-panel>
</bk-tabs>
`;

exports[`<bk-tabs/> should change tab on click: snapshot_tab_2_click 1`] = `
<bk-tabs aria-orientation="horizontal" class="bk-tabs bk-tabs--simple bk-tabs--top hydrated">
  <!---->
  <div class="bk-tabs__header">
    <div class="bk-tabs__scroll">
      <div class="bk-tabs__nav" role="tablist">
        <div class="bk-tabs__active-bar" style="transform: translateX(74.1406px); width: 34.25px;"></div>
        <div slot="header">
          <bk-tab-header aria-controls="pane_tab_1" aria-selected="false" class="bk-tab-header hydrated" id="tab_1" role="tab" tab="tab_1" tabindex="-1">
            <!---->
            <div data-tab-header>
              Tab 1
            </div>
          </bk-tab-header>
          <bk-tab-header active="" aria-controls="pane_tab_2" aria-selected="true" class="bk-tab-header hydrated is-active" id="tab_2" role="tab" tab="tab_2" tabindex="0">
            <!---->
            <div data-tab-header>
              Tab 2
            </div>
          </bk-tab-header>
        </div>
      </div>
    </div>
  </div>
  <bk-tab-panel aria-labelledby="tab_1" class="bk-tab-panel hydrated" id="pane_tab_1" role="tabpanel" tab="tab_1" style="display: none;">
    <!---->
    Content 1
  </bk-tab-panel>
  <bk-tab-panel aria-labelledby="tab_2" class="bk-tab-panel hydrated" id="pane_tab_2" role="tabpanel" tab="tab_2">
    <!---->
    Content 2
  </bk-tab-panel>
</bk-tabs>
`;

exports[`<bk-tabs/> should render: snapshot_overall 1`] = `
<bk-tabs aria-orientation="horizontal" class="bk-tabs bk-tabs--simple bk-tabs--top hydrated">
  <!---->
  <div class="bk-tabs__header">
    <div class="bk-tabs__scroll">
      <div class="bk-tabs__nav" role="tablist">
        <div class="bk-tabs__active-bar" style="transform: translateX(0px); width: 34.25px;"></div>
        <div slot="header">
          <bk-tab-header active="" aria-controls="pane_tab_1" aria-selected="true" class="bk-tab-header hydrated is-active" id="tab_1" role="tab" tab="tab_1" tabindex="0">
            <!---->
            <div data-tab-header>
              Tab 1
            </div>
          </bk-tab-header>
          <bk-tab-header aria-controls="pane_tab_2" aria-selected="false" class="bk-tab-header hydrated" id="tab_2" role="tab" tab="tab_2" tabindex="-1">
            <!---->
            <div data-tab-header>
              Tab 2
            </div>
          </bk-tab-header>
        </div>
      </div>
    </div>
  </div>
  <bk-tab-panel aria-labelledby="tab_1" class="bk-tab-panel hydrated" id="pane_tab_1" role="tabpanel" tab="tab_1">
    <!---->
    Content 1
  </bk-tab-panel>
  <bk-tab-panel aria-labelledby="tab_2" class="bk-tab-panel hydrated" id="pane_tab_2" role="tabpanel" tab="tab_2" style="display: none;">
    <!---->
    Content 2
  </bk-tab-panel>
</bk-tabs>
`;
